src/components/Item.js

```diff
@@ -0,0 +1,72 @@
+import React, { useEffect, useState } from 'react';
+import '../components/AddItemForm.css';
+import { updateDoc, serverTimestamp, doc } from 'firebase/firestore';
+import { db } from '../lib/firebase';
+import './Item.css';
+
+function Item({ item, i }) {
+  const [checked, setChecked] = useState(false);
+  const [daysSincePurchased, setDaysSincePurchased] = useState();
+  const userToken = window.localStorage.getItem('userToken');
+
+  useEffect(() => {
+    if (item.lastPurchased) {
+      const lastPurchasedSeconds = item.lastPurchased.seconds;
+      const dateNowSeconds = Date.now() / 1000;
+      const differenceInSeconds = dateNowSeconds - lastPurchasedSeconds;
+
+      setDaysSincePurchased(differenceInSeconds / 86400);
+      if (differenceInSeconds / 86400 < 1) {
+        setChecked(true);
+      }
+    }
+  }, [item]);
+
+  useEffect(() => {
+    if (daysSincePurchased > 1) {
+      if (checked) {
+        setChecked(false);
+      }
+    }
+  }, [daysSincePurchased, checked]);
+
+  const handleCheckboxChange = () => {
+    setChecked(!checked);
+  };
+
+  const updateLastPurchased = async (event) => {
+    const docRef = doc(db, 'users', `${userToken}`, 'list', item.id);
+    if (event.target.checked) {
+      await updateDoc(docRef, {
+        lastPurchased: serverTimestamp(),
+      });
+    } else {
+      await updateDoc(docRef, {
+        lastPurchased: null,
+      });
+    }
+  };
+
+  return (
+    <div>
+      <div className="item">
+        <label htmlFor="itemPurchased">Purchased</label>
+        <form>
+          <input
+            id="itemPurchased"
+            type="checkbox"
+            checked={checked}
+            name="itemPurchased"
+            onChange={handleCheckboxChange}
+            onClick={updateLastPurchased}
+          />
+        </form>
+        <ul className="item-name" i={i}>
```

Comment:
I would recommend naming this `key` instead of `i`.
```suggestion
        <ul className="item-name" key={i}>
```